src/main/java/org/osiam/auth/login/internal/InternalAuthenticationProvider.java

```diff
@@ -104,8 +107,13 @@ public Authentication authenticate(Authentication authentication) {
             throw new DisabledException("The user with the username '" + username + "' is disabled!");
         }
 
-        if (!passwordEncoder.isPasswordValid(user.getPassword(), password, user.getId())) {
-            throw new BadCredentialsException("Bad credentials");
+        if (!bCryptPasswordEncoder.matches(password, user.getPassword())) {
+            if (passwordEncoder.isPasswordValid(user.getPassword(), password, user.getId())) {
+                User replaceUser = new User.Builder(user).setPassword(password).build();
+                userProvisioning.replace(user.getId(), replaceUser);
+            } else {
```

Comment:
Then the `BadCredentialsException` would be thrown when the password matches with the bcrypt password encoder.
The `BadCredentialsException` is thrown when both encoders do not return with true.